### ISSUE TITLE
authentication via token

### ISSUE THREAD

**Issue by andischmid** [MarioRuiz/nice_http#10]
I'm using your gem to connect to a server using a token.
I found that it only works if I use **strict_encode64** instead of your util method that uses **encode64**.

No idea if this applies in general or just for my tokens though...

best regards
andi
ps: many thanks for your great and clean work!

**Comment by MarioRuiz** [comment]
I didn't see any issues in other basic_authentication tokens anyway this could be the reason: https://stackoverflow.com/a/15301870/2546667
Thanks for the words!

**Comment by andischmid** [comment]
Hi Mario
thanks for the quick reply. 
I guess I found the issue... `encode64` will add a LF at the end of the payload. If your string (user_name and token) to be encoded is longer than 44 characters then it even adds an additional LF after that character. This might not be accepted by the server so it refuses the token. `strict_encode64` doesn't add any LFs at all.
try
```
puts Base64.encode64('0123456789012345678901234567890123456789012345').bytes
puts Base64.strict_encode64('0123456789012345678901234567890123456789012345').bytes

```

**Comment by MarioRuiz** [comment]
I'll change the method in utils I think it makes sense.
Thanks!!!

**Comment by MarioRuiz** [comment]
released 1.7.14 allowing an option strict: true so both ways will be allowed.
Thanks @andischmid for your contribution!
